Sources/SwiftParser/Expressions.swift

```diff
@@ -1704,29 +1704,28 @@ extension Parser {
           let specifier = self.parseClosureCaptureSpecifiers()
 
           // The thing being capture specified is an identifier, or as an identifier
-          // followed by an expression.
+          // followed by an initializer clause.
           let unexpectedBeforeName: RawUnexpectedNodesSyntax?
-          let name: RawTokenSyntax?
-          let unexpectedBeforeEqual: RawUnexpectedNodesSyntax?
-          let equal: RawTokenSyntax?
-          let expression: RawExprSyntax
+          let name: RawTokenSyntax
+          let initializer: RawInitializerClauseSyntax?
```

Comment:
You should be able to hoist the parsing of `name` out of the `if` statement and then use `self.at(.equal)` instead of `self.peek(isAt: .equal)`

Reply:
Thanks for the suggestion. I moved name parsing to before the `if` statement.

As an interesting artifact, `self` token will not be remapped to `identifier` when parsing capture of `[self = 123]` form. Out of curiosity, was there a particular reason for this remapping? As far as I checked, e.g. in `let self = 123` `self` is classified as a keyword.